src/config.ts

```diff
@@ -1,17 +1,32 @@
-import type { VitePWAOptions } from 'vite-plugin-pwa'
+import type { DefaultTheme, UserConfig } from 'vitepress'
+import { escapeStringRegexp } from './utils'
 
-export function configurePWAOptions(options: Partial<VitePWAOptions>) {
-  if (!options.outDir)
-    options.outDir = '.vitepress/dist'
+export function configurePWAOptions<T = DefaultTheme.Config>(config: UserConfig<T>) {
+  const pwa = config.pwa ?? {}
+  const assetsDir = config.assetsDir
+    ? config.assetsDir
+      .replace(/^\.?\/|\/$/g, '')
+      .replace(/\\/g, '/')
```

Comment:
```suggestion
      .replace(/\\/g, '/')
      .replace(/^\.?\/|\/$/g, '')
```

slash normalization should be done before to handle things like `.\foo`